code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm

```diff
@@ -316,8 +316,9 @@
 /datum/reagent/toxin/fertilizer/monoammoniumphosphate/touch_mob(var/mob/living/L, var/amount)
 	. = ..()
 	if(istype(L))
-		L.ExtinguishMob(L.on_fire ? amount*3 : amount*1.5)
-		remove_self(amount)
+		var/needed = L.fire_stacks >= 3 * amount ? 3 * amount : L.fire_stacks
+		L.ExtinguishMob(needed)
+		remove_self(needed)
```

Comment:
uuuuuh. If this ends up being that `needed = 3 * amount` then we're going to remove more than the amount. 🤔 Though I don't think it should break something. But something to consider.

Reply:
Resolved